src/OptimalTransport.jl

```diff
@@ -192,7 +199,11 @@ function sinkhorn(mu, nu, C, eps; kwargs...)
     # compute dual potentials
     u, v = sinkhorn_gibbs(mu, nu, K; kwargs...)
 
-    return Diagonal(u) * K * Diagonal(v)
+    if mu isa Vector
```

Comment:
It seems a bit too restrictive, maybe `AbstractVector` would be better
```suggestion
    if mu isa AbstractVector
```
or
```suggestion
    if ndims(mu) == 1
```
(this can be inferred from the type of `AbstractArray`).